reactioncommerce/reaction
# feat: 4106 nnnnat catalogItem GQL query

Resolves #4106   
Impact: **minor**  
Type: **feature**

## Issue
**Summary**
Create a query to retrieve a single catalog product by the provided slug or ID.

**Implementation**
The catalogProduct query should have the following signature

```
catalogProduct(slugOrId: String): CatalogItemProduct
```

See https://github.com/reactioncommerce/reaction/issues/4105 for schema detail.

**Acceptance Criteria**
Using GraphQL, I can use the catalogProduct(slugOrId: String): CatalogItemProduct query to get a single CatalogProduct by it's slug.

## Solution
* Created a Catalog Collection Query that returns the product from the catalog collection
* Created a catalogItemProduct GQL query that returns a CatalogItemProduct object
* Created unit test for both of the new query functions
* Created integration test for the CatalogItemProduct GQL query. 

## Breaking changes
N/A

## Testing
1. Spin up a fresh RC shop, login as an admin and create a few products and publish them to the catalog.
    *  **Note:** if you use the reaction devtools to speed products you'll need to update your RC shop name to be "reaction swag shop" or you'll get an error from reaction. You may need to change the shop name from the DB because of this (#4202) issue.
2. From graphiql or localhost:3030/graphiql query a product by slug with `catalogItemProduct(slugOrId: "example-product") { _id }` and copy the returned ID
3. Try the query again this time with the ID instead of the slug `catalogItemProduct(slugOrId: "opaqueID") { _id product { slug } }`
4. The same product should return.
